Return empty strings as empty string

brendanlong opened the pull request:
Previously, they were returned as NULL.

brendanlong commented on test/test_dblib.ml:
```diff
@@ -35,17 +35,27 @@ let test_connect ((_, _, _, database) as params) _ =
-      Dblib.sqlexec conn "SELECT CAST(1 AS INT) AS test";
+      Dblib.sqlexec conn
+        "SELECT CAST(1 AS INT) AS test, CAST('' AS VARCHAR(10)) AS str, \
+         CAST(NULL AS BIT) AS n";
```
Note: The casts here are just to avoid different DB versions returning these as different kinds of int, str, etc. (And the return type of a NULL seems completely arbitrary).

Chris00 replied:
OK.  I think that comment should be in the file itself.

brendanlong commented on src/dblib_stubs.c:
```diff
@@ -389,7 +389,7 @@ CAMLexport value ocaml_freetds_dbnextrow(value vdbproc)
       len = dbdatlen(dbproc, c); /* length, in bytes, of the data for
                                     a column. */
-      if (len == 0 || data == NULL) {
+      if (data == NULL) {
```
http://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.help.ocs_12.5.1.dblib/html/dblib/X69467.htm

> Returns
> A BYTE pointer to the data for the particular column of interest. Be sure to cast this pointer into the proper type. A NULL BYTE pointer is returned if there is no such column or if the data has a null value. To make sure that the data is really a null value, you should always check for a return of 0 from dbdatlen.

So technically we should distinguish `data == NULL && len == 0` -> NULL and `data == NULL && len < 0` -> column doesn't exist.

brendanlong commented:
I changed this to throw a fatal exception if `len == -1`, or if `data == NULL && len != 0`. I also added more tests.

And I noticed there was an unconditional `exit(0)` and replaced it with a raise_fatal, but I can take that commit back out if you don't like it.

brendanlong commented:
I'm trying to debug an issue where on one specific platform (Amazon Linux),  I'm getting `" "` returned instead of `""` :\

Chris00 commented:
Since I do not have anymore access to a TDS server — maybe I should install one on my machine... — I set up AppVeyor to perform the tests.  I can [reproduce your problem](https://ci.appveyor.com/project/Chris00/ocaml-freetds#L499).  Note that the length returned by `dbdatlen` is 1 for the 3 problematic columns...

Chris00 commented:
I already spent way too much time on this.  I think I'll push the tests to the main branch and open an issue about this... — for now, the code is at https://github.com/Chris00/ocaml-freetds

Chris00 commented:
In 2006, some people said [so the problem is.... dblib! Using dblib is impossible to distingue between empty string and NULL (in dblib a field is NULL if length/size is 0) so the db used to return a space (' ') for empty strings however our internal library support empty string but dblib do not consider that empty string
should be converted to single space string in order to avoid '' -> NULL](https://sourceforge.net/p/freetds/bugs/25/#4850).  Would it be possible that it persists until now?!!

Chris00 commented:
Conversation continued on https://github.com/kennknowles/ocaml-freetds/issues/24

Chris00 commented:
@brendanlong If you have a bit of time, may you write tests for `Ct`?  Maybe this library is more sane.  I have read that [Dblib is deprecated](http://www.databaseteam.org/5-sybase/53b460b19de53cd7.htm).
